### PR TITLE
Right to Left UI support

### PR DESCRIPTION
**Pull request by mXaln**
This change is [Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/416)

### CODE REVIEW

**Comment by mXaln** [comment]
In this PR I excluded audio players and waveforms from RTL mirroring, because it's a bad practice according to Material Design recommendations https://material.io/design/usability/bidirectionality.html

**Review by jsarabia — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/416)* status: 0 of 57 files reviewed, 1 unresolved discussion (waiting on @jsarabia and @mXaln)

___
*[jvm/markerapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/markerapp/app/view/SourceTextFragment.kt, line 46 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/416#-Mnk7hg_0rKvjMHcOdxQ:-Mnk7hg_0rKvjMHcOdxR:b-3m5gup) ([raw file](https://github.com/bible-translation-tools/orature/blob/c66fcf8f3d0cc84f6aaa6651dff894b22b64f12c/jvm/markerapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/markerapp/app/view/SourceTextFragment.kt#L46)):*
> ```Kotlin
>                 license = parameters.named["license"]
>                 direction = parameters.named["direction"]
>                 sourceDirection = parameters.named["source_direction"]
> ```

can you make these default to ltr if not provided?

**Review by mXaln — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/416)* status: 0 of 57 files reviewed, 1 unresolved discussion (waiting on @jsarabia)

___
*[jvm/markerapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/markerapp/app/view/SourceTextFragment.kt, line 46 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/416#-Mnk7hg_0rKvjMHcOdxQ:-Mnkf4cT76xUCVuY5ocK:bp6cjke) ([raw file](https://github.com/bible-translation-tools/orature/blob/c66fcf8f3d0cc84f6aaa6651dff894b22b64f12c/jvm/markerapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/markerapp/app/view/SourceTextFragment.kt#L46)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

can you make these default to ltr if not provided?
</blockquote></details>

It does that already. I if check the code below, it will be RTL only when parameter is exactly "rtl", otherwise if it's "ltr" or any or NULL, it will be LTR.
```
when (direction) {
    "rtl" -> NodeOrientation.RIGHT_TO_LEFT
     else -> NodeOrientation.LEFT_TO_RIGHT
}
```

**Review by jsarabia — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/416)* status: 0 of 57 files reviewed, 1 unresolved discussion (waiting on @jsarabia and @mXaln)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/WorkbookDataStore.kt, line 99 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/416#-MnliRVP5UOmp8zN2BAe:-MnliRVQAJfDfYgF1G9a:b-ip5nob) ([raw file](https://github.com/bible-translation-tools/orature/blob/c66fcf8f3d0cc84f6aaa6651dff894b22b64f12c/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/WorkbookDataStore.kt#L99)):*
> ```Kotlin
>             else -> NodeOrientation.LEFT_TO_RIGHT
>         }
>     }
> ```

These don't belong in the workbook data store. The workbook data store is meant like a flux type store for state specifically regarding user selections of workbook navigation. Think Redux.

I would rather these go in the root view viewmodel, or some kind of app preferences store (which I guess the settings viewmodel is becoming)

**Review by mXaln — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/416)* status: 0 of 58 files reviewed, 1 unresolved discussion (waiting on @jsarabia)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/WorkbookDataStore.kt, line 99 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/416#-MnliRVP5UOmp8zN2BAe:-MnzLnMd5Bd3jlxq-NDw:bqs4mb7) ([raw file](https://github.com/bible-translation-tools/orature/blob/c66fcf8f3d0cc84f6aaa6651dff894b22b64f12c/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/WorkbookDataStore.kt#L99)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

These don't belong in the workbook data store. The workbook data store is meant like a flux type store for state specifically regarding user selections of workbook navigation. Think Redux.

I would rather these go in the root view viewmodel, or some kind of app preferences store (which I guess the settings viewmodel is becoming)
</blockquote></details>

Done. Moved to the SettingsViewModel

**Review by jsarabia — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/416#-:-MoKQWxf5AbOfLmYvtvw:bnfp4nl)* status: 0 of 58 files reviewed, 1 unresolved discussion (waiting on @jsarabia)

File: jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/AudioPluginViewModel.kt [hunk 1]
```diff
@@ -137,7 +142,10 @@ class AudioPluginViewModel : ViewModel() {
     }
 
     fun addPlugin(record: Boolean, edit: Boolean) {
-        find<AddPluginDialog>().open()
+        find<AddPluginDialog>().apply {
+            orientationProperty.set(settingsViewModel.orientationProperty.value)
```
Comment (AnonymousWalker) [inline jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/AudioPluginViewModel.kt:146]: @mXaln shouldn't this use "bind" instead?

**Pull request merged**